deploy/examples/create-external-cluster-resources.py

```diff
@@ -565,31 +543,113 @@ def create_cephCSIKeyring_cephFSNode(self):
-                            "format": "json"}
+                caps = {"mon": "allow r, allow command 'osd blocklist'",
+                        "mgr": "allow rw",
+                        "osd": "allow rw tag cephfs data={}".format(cephfs_filesystem),
+                        "mds": "allow rw"}
+        else:
+            caps = {"mon": "allow r, allow command 'osd blocklist'",
+                    "mgr": "allow rw",
+                    "osd": "allow rw tag cephfs *=*",
+                    "mds": "allow rw"}
+            
+        return caps,entity    
+    
+    def get_rbd_provisioner_caps_and_entity(self):
+        entity = "client.csi-rbd-provisioner"
+        caps = {}
+        if self._arg_parser.restricted_auth_permission:
+            rbd_pool_name = self._arg_parser.rbd_data_pool_name
+            cluster_name = self._arg_parser.cluster_name
+            rados_namespace = self._arg_parser.rados_namespace
+            if rbd_pool_name == "" or cluster_name == "" or rados_namespace == "":
+                raise ExecutionFailureException(
+                    "mandatory flags not found, please set the '--rbd-data-pool-name', '--cluster-name' and --rados-namespace flags")
+            entity = "client.csi-rbd-provisioner-{}-{}-{}".format(cluster_name,rbd_pool_name,rados_namespace)
+            caps = {"mon": "profile rbd, allow command 'osd blocklist'",
+                    "mgr": "allow rw",
+                    "osd": "profile rbd pool={}".format(rbd_pool_name)}
         else:
-            cmd_json = {"prefix": "auth get-or-create",
+            caps = {"mon": "profile rbd, allow command 'osd blocklist'",
+                    "mgr": "allow rw",
+                    "osd": "profile rbd"}
+        
+        return caps,entity    
+    
+    def get_rbd_node_caps_and_entity(self):
+        entity = "client.csi-rbd-node"
+        caps = {}
+        if self._arg_parser.restricted_auth_permission:
+            rbd_pool_name = self._arg_parser.rbd_data_pool_name
+            cluster_name = self._arg_parser.cluster_name
+            rados_namespace = self._arg_parser.rados_namespace
+            if rbd_pool_name == "" or cluster_name == "" or rados_namespace == "":
+                raise ExecutionFailureException(
+                    "mandatory flags not found, please set the '--rbd-data-pool-name', '--cluster-name' and --rados-namespace flags")
+            entity = "client.csi-rbd-node-{}-{}-{}".format(cluster_name,rbd_pool_name,rados_namespace)
+            caps = {"mon": "profile rbd, allow command 'osd blocklist'",
+                    "osd": "profile rbd pool={}".format(rbd_pool_name),}
+        else:
+            caps = {"mon": "profile rbd, allow command 'osd blocklist'",
+                    "osd": "profile rbd"}
+        
+        return caps,entity
+    
+    def get_caps_and_entity(self, user_name):
+        if "client.csi-cephfs-provisioner" in user_name:
+            if "client.csi-cephfs-provisioner" != user_name:
```

Comment:
@leseb figured the way that the script will auto-detect if the user is auth restricted or not.

But if we upgrade restricted auth user then, we need to provide --rbd-data-pool-name, --rados-namespace,
 --cluster-name flag while upgrading.
 As they are variable always with restricted auths.

If you say I can fetch the above flag details from the user name, but that doesn't seem robust in terms of validation.

Reply:
I think the `--upgrade` might require the `--rbd-data-pool-name` already. So if we detect the user is restricted let's ask the user the missing flag (rados/cluster-name), also let's be careful since the rados namespace is optional AFAIR.

Reply:
Yes right,
The `--rbd-data-pool-name, --rados-namespace and --cluster-name` are required, fs-name can be optional:)